bioptim/limits/objective_functions.py

```diff
@@ -96,15 +96,25 @@ def print(self):
         raise NotImplementedError("Printing of ObjectiveList is not ready yet")
 
     @staticmethod
-    def get_nlp_objectives(nlp):
-        list_objectives = [[{"Mayer": [], "Lagrange": []} for _ in range(NLP.ns + 1)] for NLP in nlp]
-        for NLP in nlp:
-            for J in NLP.J:
+    def get_nlp_objectives(all_nlp):
+        list_objectives = [[{"Mayer": [], "Lagrange": [], "Quadratic_Mayer": [], "Quadratic_Lagrange": [], "Sliced_target_Mayer": [], "Sliced_target_Lagrange": [], "Parameters_Mayer": [], "Parameters_Lagrange": []} for _ in range(nlp.ns + 1)] for nlp in all_nlp]
+        for nlp in all_nlp:
+            for J in nlp.J:
                 for n in J:
                     if isinstance(n["objective"].type, ObjectiveFcn.Lagrange):
-                        list_objectives[NLP.phase_idx][n["node_index"]]["Lagrange"].append(n["objective"].name)
+                        list_objectives[nlp.phase_idx][n["node_index"]]["Lagrange"].append(n["objective"].name)
+                        list_objectives[nlp.phase_idx][n["node_index"]]["Quadratic_Lagrange"].append(n["objective"].quadratic)
+                        list_objectives[nlp.phase_idx][n["node_index"]]["Sliced_target_Lagrange"].append(
```

Comment:
This is a way to simplify display task. I guess it seems relevant to directly separate data related to objectives by type. Indeed, the way Lagrange/Mayer are displayed in the final graph are different (see display functions in ocp.py). If we don’t separate them by type directly, we would need to use one more index, and I guess it could complexify debug as well as the organization of the code (especially for display functions in ocp.py).

Reply:
Done